# Intervalos de confiança 

In [1]:
import numpy as np
from scipy.stats import norm
from scipy import stats
import seaborn as sns
import math

In [2]:
dados = np.array([126. , 129.5, 133. , 133. , 136.5, 136.5, 140. , 140. , 140. ,
                  140. , 143.5, 143.5, 143.5, 143.5, 143.5, 143.5, 147. , 147. ,
                  147. , 147. , 147. , 147. , 147. , 150.5, 150.5, 150.5, 150.5,
                  150.5, 150.5, 150.5, 150.5, 154. , 154. , 154. , 154. , 154. ,
                  154. , 154. , 154. , 154. , 157.5, 157.5, 157.5, 157.5, 157.5,
                  157.5, 157.5, 157.5, 157.5, 157.5, 161. , 161. , 161. , 161. ,
                  161. , 161. , 161. , 161. , 161. , 161. , 164.5, 164.5, 164.5,
                  164.5, 164.5, 164.5, 164.5, 164.5, 164.5, 168. , 168. , 168. ,
                  168. , 168. , 168. , 168. , 168. , 171.5, 171.5, 171.5, 171.5,
                  171.5, 171.5, 171.5, 175. , 175. , 175. , 175. , 175. , 175. ,
                  178.5, 178.5, 178.5, 178.5, 182. , 182. , 185.5, 185.5, 189., 192.5])

In [3]:
N = len(dados)
media = np.mean(dados)
std = np.std(dados)

In [4]:
alpha = (1 - 0.95)/2

In [6]:
1 - alpha # Esse valor vamos buscar na tabela z-score e colocar na fórmula depois

0.975

O valor na tabela fica 1.95

In [7]:
# usando o scipy sem precisar consultar a tabela
z = norm.ppf(1-alpha)
z

1.959963984540054

In [9]:
x_inferior = media - z * (std/math.sqrt(N))
x_superior = media + z * (std/math.sqrt(N))

In [10]:
x_inferior, media, x_superior

(156.57464916110283, 159.25, 161.92535083889717)

In [12]:
# Calculo da margem de erro:

margem_erro = abs(media-x_superior)
margem_erro

2.675350838897174

Logo, podemos determinar que temo 95% de confiança de que a media das altura das pessoas está no intervalo: 156.57, 159.25, 161.92

# Utilizando o SCIPY

In [13]:
stats.sem(dados) # é igual a isso std/math.sqrt(N)
intervalos = norm.interval(0.95, media, stats.sem(dados))

In [14]:
intervalos

(156.5611712378229, 161.9388287621771)

In [15]:
margem_erro = media - intervalos[0]
margem_erro

2.68882876217711

# Diferentes níveis de confiança 

In [17]:
intervalos = norm.interval(0.99, media, stats.sem(dados))
intervalos

(155.71628000694295, 162.78371999305705)

Quando maior a cofiança maior será a diferença entre os intervalos.

In [18]:
margem_erro = media - intervalos[0]
margem_erro

3.5337199930570478

Margem de erro maior. 

In [19]:
intervalos = norm.interval(0.8, media, stats.sem(dados))
intervalos

(157.49186937268743, 161.00813062731257)

In [20]:
margem_erro = media - intervalos[0]
margem_erro

1.7581306273125676

Isso quer dizer que a margem de erro é menor, pois chega mais próximo do valor da média da amostra, entretanto, não retrata bem, pois as chances de algum dado cair fora desse escopo é possível